cupy/core/core.pyx

```diff
@@ -1320,13 +1320,12 @@ cdef class ndarray:
                 and slices == slice(None, None, None)
                 and isinstance(value, numpy.ndarray)
         ):
-            if self.dtype == value.dtype and self.shape == value.shape:
-                if self.strides == value.strides:
-                    ptr = ctypes.c_void_p(value.__array_interface__['data'][0])
-                else:
-                    order = 'F' if self.flags.f_contiguous else 'C'
-                    tmp = value.ravel(order)
-                    ptr = ctypes.c_void_p(tmp.__array_interface__['data'][0])
+            if (self.dtype == value.dtype
+                    and self.shape == value.shape
+                    and (self.flags.f_contiguous or self.flags.c_contiguous)):
+                order = 'F' if self.flags.f_contiguous else 'C'
```

Comment:
Use CuPy's internal flags:
```suggestion
                    and (self._f_contiguous or self._c_contiguous)):
                order = 'F' if self._f_contiguous else 'C'
```

Reply:
What is the benefit of using internal flags? (First time I see the "Suggested change" feature of github, really cool!)

Reply:
Performance. It's faster to access the C attributes than the Python attributes, so for array metadata like contiguity and shape we use internal C attributes whenever possible in the Cython space.

Reply:
Smart! Applied. Thanks for the tip and the review so far!